panel/widgets/chat.py

```diff
@@ -0,0 +1,1328 @@
+    auto_send_types : Tuple[Widget]
+        The widget types to automatically send when the user presses enter
+        or clicks away from the widget. If not provided, defaults to
+        `[TextInput]`.
+    user : str
+        Name of the ChatInterface user.
+    avatar : str | BinaryIO
+        The avatar to use for the user. Can be a single character text, an emoji,
+        or anything supported by `pn.pane.Image`. If not set, uses the
+        first character of the name.
+    reset_on_send : bool
+        Whether to reset the widget's value after sending a message;
+        has no effect for `TextInput` and non string-based inputs.
+    show_send : bool
+        Whether to show the send button.
+    show_rerun : bool
+        Whether to show the rerun button.
+    show_undo : bool
+        Whether to show the undo button.
+    show_clear : bool
+        Whether to show the clear button.
+    show_button_name : bool
+        Whether to show the button name.
+    """
+
+    widgets = param.List(doc="""
+        Widgets to use for the input. If not provided, defaults to
+        `[TextInput]`.""")
+
+    auto_send_types = param.List(doc="""
+        The widget types to automatically send when the user presses enter
+        or clicks away from the widget. If not provided, defaults to
+        `[TextInput]`.""")
+
+    user = param.String(default="User", doc="Name of the ChatInterface user.")
+
+    avatar = param.ClassSelector(class_=(str, BinaryIO), doc="""
+        The avatar to use for the user. Can be a single character text, an emoji,
+        or anything supported by `pn.pane.Image`. If not set, uses the
+        first character of the name.""")
+
+    reset_on_send = param.Boolean(default=False, doc="""
+        Whether to reset the widget's value after sending a message;
+        has no effect for `TextInput`.""")
+
+    show_send = param.Boolean(default=True, doc="""
+        Whether to show the send button.""")
+
+    show_rerun = param.Boolean(default=True, doc="""
+        Whether to show the rerun button.""")
+
+    show_undo = param.Boolean(default=True, doc="""
+        Whether to show the undo button.""")
+
+    show_clear = param.Boolean(default=True, doc="""
+        Whether to show the clear button.""")
+
+    show_button_name = param.Boolean(default=True, doc="""
+        Whether to show the button name.""")
+
+    _input_container = param.ClassSelector(class_=Row, doc="""
+        The input message row that wraps the input layout (Tabs / Row)
+        to easily swap between Tabs and Rows, depending on
+        number of widgets.""")
+
+    _input_layout = param.ClassSelector(class_=(Row, Tabs), doc="""
+        The input layout that contains the input widgets.""")
+
+    _button_data = param.Dict(default={}, doc="""
+        Metadata and data related to the buttons.""")
+
+    _stylesheets: ClassVar[List[str]] = [
+        f"{CDN_DIST}css/chat_interface.css"
+    ]
+
+    def __init__(self, **params):
```

Comment:
I believe it would help the user if he/ she was also able to provide a single widget directly, i.e. not having to wrap it in a list.

It shortens the code you have to type. And it lowers the probability of the user experiencing an exception.

```python
ChatInterface(
    ....,
    widgets=pn.widgets.TextInput(...)
)
```

Reply:
Implemented.